packages/engine-svg/index.js

```diff
@@ -1,40 +1,44 @@
-import { Mechanic } from "@mechanic-design/core";
+import { Mechanic } from '@mechanic-design/core';
 
-const root = document.getElementById("root");
-const head = document.querySelector("head");
+const root = document.getElementById('root');
+const head = document.querySelector('head');
 
 export const run = (functionName, func, values, config) => {
   const { isPreview } = config;
 
-  root.innerHTML = "";
+  root.innerHTML = '';
 
   const mechanic = new Mechanic(func.settings, values, config);
-  const onFrame = (el) => {
-    // Pending virtual-dom approach
-    root.innerHTML = el.trim();
-    if (!isPreview) {
-      mechanic.frame(root.childNodes[0], { head });
-    }
-  };
-  const onDone = async (el, name) => {
-    root.innerHTML = el.trim();
-    if (!isPreview) {
-      await mechanic.done(root.childNodes[0], { head });
-      mechanic.download(name || functionName);
-    }
-  };
-  const onSetState = async (obj) => {
-    mechanic.setState(obj);
-  };
 
-  func.handler({
-    inputs: mechanic.values,
-    mechanic: {
-      frame: onFrame,
-      done: onDone,
-      state: mechanic.functionState,
-      setState: onSetState,
+  mechanic.dispatch({
```

Comment:
I feel like I might have taken the abstraction too far here. But I kind of like having this centralized entrypoint for engines.

Not sure about the object that's passed to dispatch. I could also imagine this could be clean by registering the callbacks and then move `frameOrDone` also to core.

```js
mechanic.registerFrameCallback((el) => ...);
mechanic.registerDoneCallback((el) => ...);
```

We need to keep a reference of each engine's helpers in the core at runtime so the automatic return value frame capturing for `static` and `animation` works using `maybeAddFrame`.

Let me know what you think :blush:

Reply:
I just had some time and tried rewriting this to use register style functions. I think it's much more readable like this and provides a nice shared way to add the information to core that core needs to correctly control the draw loop